include/deal.II/lac/vector.templates.h

```diff
@@ -2048,8 +2045,12 @@ Vector<Number>::allocate()
 {
   // make sure that we don't create a memory leak
   Assert (val == 0, ExcInternalError());
-  val = static_cast<Number *>(_mm_malloc (sizeof(Number)*max_vec_size, 64));
-  Assert (val != 0, ExcOutOfMemory());
+
+  // then allocate memory with the proper alignment requirements of 64 bytes
+  const int ierr = posix_memalign ((void **)&val, 64, sizeof(Number)*max_vec_size);
+
```

Comment:
Didn't we only include mm_malloc.h if we were not on _MSC_VER? I check by myself and it seems that visual studio provides `_aligned_malloc` (see e.g. here: https://msdn.microsoft.com/en-us/library/8z34s9c6(VS.80).aspx)

However, given how rare the use of visual studio is, we should simply stick with a standard malloc in that case. This way, we can use the standard 'free' methods as before.